src/mame/machine/k573enc.h

```diff
@@ -0,0 +1,120 @@
+// license:BSD-3-Clause
+// copyright-holders:windyfairy
+#ifndef MAME_MACHINE_K573ENC_H
+#define MAME_MACHINE_K573ENC_H
+
+#pragma once
+
+#include <map>
+
+std::unordered_map<uint16_t, uint32_t> k573enc_lookup({
```

Comment:
What is that table supposed to be?  There is no such thing in the fpga.

Reply:
This ties directly into the other comment so I'll respond to both here.

>Do you mean that ddrsbm has a different fpga firmware?

ddrsbm is the only game on the k573dio platform that uses a different encryption algorithm compared to everything else. The fake FPGA algorithm is an approximation of the FPGA algorithm that, when given a decryption key and scramble key and counter (ddrsbm doesn't use a counter though), you can generate the same decrypted data that the real algorithm would provide.

ddrsbm is also the only game that uses FPGA key 1 only as its input. Every other game uses key parts 2 and 3. Its scramble key is also always derived from the decryption key, whereas the scramble key would need to be something completely different from the decryption key for every other game's data.

The table itself was derived using real hardware to get the plain text data using the real FPGA firmware, and then calculating the decryption key based on the plaintext data. With the lack of the real algorithm being run on the FPGA for this specific game, the fake algorithm provides the same results but requires these hardcoded keys.